### PR TITLE
CYF-ITP-South Africa | Simphiwe Mabuya | Module-Structuring-and-Testing-Data | Week-2

### PR DESCRIPTION
**Pull request by Simphiwe-Mabuya**
## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

**Comment by Simphiwe-Mabuya** [comment]
@SallyMcGrath Hi Sally ..thanks a million for your feedback and for noticing the mistake I made. It really stressed me out when I saw I made that error and I didn't know how to rectify it when I made the error.  

I shall go ahead and look into the channels on slack for solutions as per your suggestion.

File: Sprint-2/errors/1.js [hunk 1]
```diff
@@ -3,11 +3,14 @@
 // Why will an error occur when this program runs?
 // Try playing computer with the example to work out what is going on
 
+// Firstly, the decimalNumber variable has already been declared. Secondly, the function convertToPercentage has not been called correctly, instead console.log was used to log decimalNumber. By removing the const keyword and calling the function correctly allows the code to convert the given decimalNumber to a percentage.
+
 function convertToPercentage(decimalNumber) {
-  const decimalNumber = 0.5;
+  decimalNumber = 0.5;
```
Comment (cjyuan) [inline Sprint-2/errors/1.js:9]: If you set the value of `decimalNumber` here, then the function will always return "50%", regardless of the value of the parameter.

File: Sprint-2/errors/1.js [hunk 2]
```diff
@@ -3,11 +3,14 @@
...
   const percentage = `${decimalNumber * 100}%`;
 
   return percentage;
 }
 
-console.log(decimalNumber);
+let myPercentage = convertToPercentage()
```
Comment (cjyuan) [inline Sprint-2/errors/1.js:15]: If the function expects a parameter, we should always pass a parameter in the function call. Otherwise the parameter will received the value `undefined`.

File: Sprint-2/extend/format-time.js [hunk 3]
```diff
@@ -2,9 +2,15 @@
...
 
 function formatAs12HourClock(time) {
-  const hours = Number(time.slice(0, 2));
+  const hours = Number(time.slice(0, 2)); // Gets the hour as a number 
+  const minutes = time.slice(3) // Gets the minutes 
+
+  if(hours === 0){                // Checks for midnight hour (00:00)
+    return `12:${minutes} am`
+  }
+
```
Comment (cjyuan) [inline Sprint-2/extend/format-time.js:11]: What should "12:01" be converted to?

File: Sprint-2/extend/format-time.js [hunk 4]
```diff
@@ -16,9 +22,31 @@ console.assert(
...
+);
+
+
+const currentOutput4 = formatAs12HourClock("14:00");
+const targetOutput4 = "02:00 pm";
+console.assert(
+  currentOutput4 === targetOutput4,
+  `current output: ${currentOutput4}, target output: ${targetOutput4}`
+);
+
```
Comment (cjyuan) [inline Sprint-2/extend/format-time.js:46]: Could also test borderline cases like "12:00", "23:59".

File: Sprint-2/implement/to-pounds.js [hunk 5]
```diff
@@ -4,3 +4,26 @@
...
+      
+      const paddedPenceNumberString = pounds.padStart(3, "0");
+      pounds = paddedPenceNumberString.substring(
+        0,
+        paddedPenceNumberString.length - 2
+      );
+      
+      const pence = paddedPenceNumberString
+        .substring(paddedPenceNumberString.length - 2)
+        .padEnd(2, "0");
```
Comment (cjyuan) [inline Sprint-2/implement/to-pounds.js:23]: Do we really need `.padEnd(2, "0")`?

File: Sprint-2/implement/vat.js [hunk 6]
```diff
@@ -8,3 +8,10 @@
 // Given a number,
 // When I call this function with a number
 // it returns the new price with VAT added on
+
+function priceWithVat(price){
+    let percentage = 1.2;
+    return `£${(price * percentage)}`;
```
Comment (cjyuan) [inline Sprint-2/implement/vat.js:14]: Between number and string types, which type do you think is more appropriate for the returned value of this function? Why?

File: Sprint-2/errors/0.js [hunk 7]
```diff
@@ -3,7 +3,12 @@
 // call the function capitalise with a string input
 // interpret the error message and figure out why an error is occurring
 
+//The error is cause by an already declared str variable. The problem is fixed by removing the let keyword.
+
 function capitalise(str) {
-  let str = `${str[0].toUpperCase()}${str.slice(1)}`;
+   str = `${str[0].toUpperCase()}${str.slice(1)}`;
```
[the remark this answers is not in the document]
Comment (Simphiwe-Mabuya) [reply Sprint-2/errors/0.js]: Refactored the code to return the value without storing it in a variable

File: Sprint-2/errors/1.js (hunk 1 above)
Comment (Simphiwe-Mabuya) [reply Sprint-2/errors/1.js:9]: Rectified the code by not setting a value for decimalNumber so that the function can return different values as passed into the parameter

File: Sprint-2/errors/1.js (hunk 2 above)
Comment (Simphiwe-Mabuya) [reply Sprint-2/errors/1.js]: Rectifies the problem ..and when I pass a parameter it doesn't return <undefined>

File: Sprint-2/extend/format-time.js (hunk 3 above)
Comment (Simphiwe-Mabuya) [reply Sprint-2/extend/format-time.js:11]: I've fixed the bug by adding another checkpoint for 12:01 pm

File: Sprint-2/extend/format-time.js (hunk 4 above)
Comment (Simphiwe-Mabuya) [reply Sprint-2/extend/format-time.js:50]: I've added assertions to test edge cases : "12:00" and "23:59".

File: Sprint-2/implement/to-pounds.js (hunk 5 above)
Comment (Simphiwe-Mabuya) [reply Sprint-2/implement/to-pounds.js:23]: We don't need padEnd() necessarily because "pence" will always have two digits. So I removed padEnd().

**Comment by Simphiwe-Mabuya** [comment]
Hi @SallyMcGrath, I managed to fix the problem you identified with assistance and guidance from @cjyuan.

**Comment by Simphiwe-Mabuya** [comment]
A good evening Mr CJ..I trust that this will find you well and in great
spirit as always.
I just wanted to check with you if I should go ahead and mark my Sprint 2
PR complete , only if there's nothing for me to fix,  for I believe I have
rectified all the concerns/questions you had.

I must thank you again for your assistance and guidance as a mentor.

Looking forward to your assistance again in the near future and hearing
from you, please do enjoy the rest of the evening.

Best regards
Simphiwe Mabuya

**Comment by cjyuan** [comment]
Unless you have doubt about your updated solutions, you can go ahead and mark your PR as "completed".
